source/opt/instruction.h

```diff
@@ -241,6 +242,18 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   // Returns true if the instruction annotates an id with a decoration.
   inline bool IsDecoration();
 
+  // Returns true if the instruction is known to be a load from read-only
+  // memory.
+  bool IsReadOnlyLoad(IRContext* ctx) const;
+
+  // Returns the instruction that gives the base address of an address
+  // calculation.
+  Instruction* GetBaseAddress(IRContext* ctx) const;
+
+  // Returns true if the instruction is a load from memory.  This does not
+  // include memory-to-memory instructions.
+  bool IsLoad() const;
```

Comment:
I prefer to keep the name to keep it close the the OpLoad instructions.  This is for OpLoad instructions or others that are similar.  I don't think if an instruction like OpCopyMemory as loads because nothing is ever visible outside of memory.  However, I have been a part of this discussion before on whether or not a memcpy like instruction is a load, so I just wanted to be explicit.

Looking at the spec, the copy operations are described as loads, for whatever that is worth.

If we want to take the position that OpCopyMemory is a load, I am fine with changing the name.